packages/flutter/lib/src/services/text_editing.dart

```diff
@@ -57,12 +57,25 @@ class TextRange {
   /// The text before this range.
   String textBefore(String text) {
     assert(isNormalized);
+    if(text.isNotEmpty && start > 0 && start < text.length ){
+      final int startCodeUnit = text.codeUnitAt(start);
+      if( startCodeUnit & 0x8C00 == 0x8C00 ){
```

Comment:
What happens if we're in a multi-byte character that has more than two bytes?  e.g. the rainbow flag emoji is four codepoints: 
```
🏳 U+1F3F3
️ U+FE0F
‍ U+200D
🌈 U+1F308
```

In testing this issue, I was able to successfully land inside those codepoints, type a character, and split the flag into a blank flag with a rainbow.  I don't think this logic would fix that scenario.

Reply:
```
main() {
  String a='🌈';
  print(a.codeUnits.map((v)=> v.toRadixString(16)));
}
```
result is (d83c, df08),    two characters,  same as '𐐷'.

flutter using UTF-16 to handle character,   

You can read the wikipedia carefully, only 2 cases, one case is 2 bytes, and the other case is 4 bytes. In flutter, it is 1 character and 2 characters. There are no other situations.

https://en.wikipedia.org/wiki/UTF-16

Reply:
I am so sorry, I'm wrong.

Multi-byte characters do exist,  like  👨‍👩‍👧 ,  🇨🇳.  Above code is not working.
I am try to find a new way.